components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/core/ParAuthServiceImpl.java

```diff
@@ -19,80 +19,109 @@
+                expiresIn, params);
     }
 
-    public Map<String, String> retrieveParams(String uuid, String clientId) throws ParCoreException {
+    @Override
+    public Map<String, String> retrieveParams(String uuid, String clientId)
+            throws ParCoreException {
+
+        Optional<ParRequestDO> optionalParRequestDO = parMgtDAO.getRequestData(uuid);
+        if (!optionalParRequestDO.isPresent()) {
+            throw new ParCoreException("A PAR request does not exist for the uuid: " + uuid);
+        }
 
-        ParRequestDO parRequestDO = parMgtDAO.getParRequest(uuid);
-        parMgtDAO.removeParRequest(uuid);
-        isRequestUriExpired(parRequestDO.getScheduledExpiryTime());
-        isClientIdValid(clientId, parRequestDO.getClientId());
+        ParRequestDO parRequestDO = optionalParRequestDO.get();
+        parMgtDAO.removeRequestData(uuid);
+        validateExpiryTime(parRequestDO.getExpiresIn());
+        validateClientID(clientId, parRequestDO.getClientId());
 
         return parRequestDO.getParams();
     }
 
-    private void isRequestUriExpired(long scheduledExpiryTime) throws ParCoreException {
+    private void validateExpiryTime(long expiresIn) throws ParClientException {
 
         long currentTimeInMillis = Calendar.getInstance(TimeZone.getTimeZone(ParConstants.UTC)).getTimeInMillis();
 
-        if (currentTimeInMillis > scheduledExpiryTime) {
-            throw new ParCoreException(OAuth2ErrorCodes.INVALID_REQUEST, "request_uri expired");
+        if (currentTimeInMillis > expiresIn) {
+            throw new ParClientException(OAuth2ErrorCodes.INVALID_REQUEST, "request_uri expired");
         }
     }
 
-    private void isClientIdValid(String clientId, String parClientId) throws
-            ParCoreException {
+    private void validateClientID(String clientId, String parClientId) throws ParClientException {
 
         if (!StringUtils.equals(parClientId, clientId)) {
-            throw new ParCoreException(OAuth2ErrorCodes.INVALID_CLIENT, "client_ids does not match");
+            throw new ParClientException(OAuth2ErrorCodes.INVALID_CLIENT,
+                    String.format("Received client_id %s does not match the client_id from the initial PAR request %s",
+                            clientId, parClientId));
         }
     }
 
-    private long getScheduledExpiry(long requestedTime) {
+    private static long getExpiresInValue() throws ParCoreException {
+
+        try {
+            Object expiryTimeValue = IdentityConfigParser.getInstance().getConfiguration().get(PAR_EXPIRY_TIME);
+            if (expiryTimeValue != null && (StringUtils.isNotBlank((String) expiryTimeValue))) {
+                return Long.parseLong((String) expiryTimeValue);
+            }
+            log.debug("PAR expiry time is not configured. Default value will be used.");
+            return ParConstants.EXPIRES_IN_DEFAULT_VALUE;
+        } catch (NumberFormatException e) {
```

Comment:
we would need to do some basic validations as well. like non-zero non-negative checks